### ISSUE TITLE
PrepareHeaderForMatch does not apply to CsvDataReader

### ISSUE THREAD

**Issue by khatran79** [JoshClose/CsvHelper#1591]
**Describe the bug**
Configuring PrepareHeaderForMatch does not apply to CsvDataReader

**To Reproduce**
```
var csvTable = new DataTable();
            using (var csvReader = new CsvReader(new StreamReader(filePath), CultureInfo.InvariantCulture))
            {
                csvReader.Configuration.PrepareHeaderForMatch = (header, index) => header.Trim()+"2";                
                CsvDataReader dr = new CsvDataReader(csvReader);
                csvTable.Load(dr);
            }
            return csvTable;
```

**Comment by gabbyhershey** [comment]
sorry I'm not sure what is the bug here?
                csvReader.Configuration.PrepareHeaderForMatch = (header, index) => header.Trim()+"2";   
This only assigns the value to be a lambda function...
Can you help me understand?

**Comment by JoshClose** [comment]
The `IDataReader.GetOrdinal(string name)` method currently ignores `PrepareHeaderForMatch`. `IDataReader.GetOrdinal` actually states this:

> GetOrdinal performs a case-sensitive lookup first. If it fails, a second case-insensitive search is made.

I'll be changing it to use `PrepareHeaderForMatch` and doing a case-insensitive match if that fails.

**Comment by JoshClose** [comment]
This is fixed in version 15.0.8 and on nuget.
